packer/configure-windows-vhd.ps1

```diff
@@ -0,0 +1,161 @@
+    }
+
+    New-Item -Path $WindowsUpdatePath | Out-Null
+    New-Item -Path $AutoUpdatePath | Out-Null
+    Set-ItemProperty -Path $AutoUpdatePath -Name NoAutoUpdate -Value 1 | Out-Null
+}
+
+
+function Get-ContainerImages
+{
+    $imagesToPull = @(
+        "mcr.microsoft.com/windows/servercore:ltsc2019",
+        "mcr.microsoft.com/windows/nanoserver:1809",
+        "mcr.microsoft.com/k8s/core/pause:1.2.0")
+
+
+    foreach ($image in $imagesToPull)
+    {
+        docker pull $image
+    }
+}
+
+function Install-Docker
+{
+    $defaultDockerVersion = "18.09"
+
+    Write-Log "Attempting to install Docker version $defaultDockerVersion"
+    Install-PackageProvider -Name DockerMsftProvider -Force -ForceBootstrap | Out-null
+    $package = Find-Package -Name Docker -ProviderName DockerMsftProvider -RequiredVersion $defaultDockerVersion
+    Write-Log "Installing Docker version $($package.Version)"
+    $package | Install-Package -Force | Out-Null
+    Start-Service docker
+}
+
+function Install-OpenSSH
+{
+    Write-Log "Installing OpenSSH Server"
+    Add-WindowsCapability -Online -Name OpenSSH.Server~~~~0.0.1.0
+}
+
+function Install-WindowsPatches
+{
+    # Windows Server 2019 update history can be found at https://support.microsoft.com/en-us/help/4464619
+    # then you can get download links by searching for specific KBs at http://www.catalog.update.microsoft.com/home.aspx
+
+    $patchUrls = @(
+        # 7C SSU and LCU
+        'http://download.windowsupdate.com/c/msdownload/update/software/secu/2019/07/windows10.0-kb4512937-x64_2a065a9ecfee76e3e457f3c596550e821358971c.msu',
+        'http://download.windowsupdate.com/d/msdownload/update/software/updt/2019/07/windows10.0-kb4505658-x64_cb660f3191eba56217694635b48d50d36883c3f2.msu')
+
+        foreach ($patchUrl in $patchUrls)
+        {
+            $pathOnly = $patchUrl.Split("?")[0]
+            $fileName = Split-Path $pathOnly -Leaf
+            $fileExtension = [IO.Path]::GetExtension($fileName)
+            $fullPath = [IO.Path]::Combine($env:TEMP, $fileName)
+
+            switch ($fileExtension)
+            {
+                ".msu"
+                {
+                    Write-Log "Downloading windows patch from $pathOnly to $fullPath"
+                    Invoke-WebRequest -UseBasicParsing $patchUrl -OutFile $fullPath
+                    Write-Log "Starting install of $fileName"
+                    $proc = Start-Process -Passthru -FilePath wusa.exe -ArgumentList "$fullPath /quiet /norestart"
+                    Wait-Process -InputObject $proc
+                    switch ($proc.ExitCode)
+                    {
+                        0
+                        {
+                            Write-Log "Finished install of $fileName"
+                        }
+                        3010
+                        {
+                            WRite-Log "Finished install of $fileName. Reboot required"
+                        }
+                        default

```

Comment:
I think we should support .exe patches too so we can test pre-release Windows patches on a private branch

Reply:
As part of this PR or can we add that later?
Also do you have an exe to test with?

Reply:
later is fine